src/main/java/com/google/devtools/build/lib/starlarkbuildapi/test/TestingModuleApi.java

```diff
@@ -56,12 +58,24 @@ ExecutionInfoApi executionInfo(Dict<?, ?> requirements // <String, String> expec
       parameters = {
         @Param(
             name = "environment",
-            named = false,
+            named = true,
             positional = true,
             doc =
                 "A map of string keys and values that represent environment variables and their"
-                    + " values. These will be made available during the test execution.")
+                    + " values. These will be made available during the test execution."),
+          @Param(
+              name = "inherited_environment",
+              allowedTypes = {@ParamType(type = Sequence.class, generic1 = String.class)},
+              defaultValue = "[]",
+              named = true,
+              positional = true,
+              doc =
+                  "A sequence of string keys that represent environment variables. These variables"
+                      + " will be made available during the test execution with their current value"
+                      + " taken from the environment. If a variable is contained in both"
+                      + " <code>environment</code> and <code>inherited_environment</code>, the"
+                      + " value inherited from the environment will take precedence if set.")
```

Comment:
> If a variable is contained in both <code>environment</code> and <code>inherited_environment</code>, the value inherited from the environment will take precedence if set.

This is the opposite of implementation in ActionEnvironment. I would also think it makes more sense that env takes precedence.

Reply:
Thanks, I modified the comment as you suggested.

> This is the opposite of implementation in ActionEnvironment. I would also think it makes more sense that env takes precedence.

Sorry, I think the missing `shell` qualifier in front of environment makes this comment confusing. Updated it to say "the value inherited from the **shell** environment will take precedence if set", which should align with the behavior of `ActionEnvironment#resolve`.
Is this the behavior you find most reasonable? I do, simply because it allows rules to choose the behavior they want: They can provide an overridable default by specifying a variable name in both `environment` and `inherited_environment`, but they can also ensure that the fixed value in `environment` is always used simply by omitting it from `inherited_environment`. If `environment` always were to take precedence, the former behavior wouldn't be possible to achieve. 

Note though that the current behavior of `env` and `env_inherit` on native rules is in fact different. [Here, `env` always takes precendence](https://github.com/bazelbuild/bazel/blob/bd35a08674fafdccf71dcc525faa2aa09824bfa7/src/main/java/com/google/devtools/build/lib/analysis/RunfilesSupport.java#L499). But since the new provider field would still allow to implement the current behavior, that shouldn't cause problems for starlarkification.

Reply:
I just noticed that `ActionEnvironment` says "The order in which the environments are combined is undefined, so callers need to take care that the key set of the {@code fixedEnv} map and the set of {@code inheritedEnv} elements are disjoint.". Is there a reason not to specify one or the other behavior?

Reply:
> Is there a reason not to specify one or the other behavior?

I think the following options are valid:
- define the behavior 
- throw an error when sets are not disjoint

Saying it's undefined and asking the user to keep it disjoint but not enforcing it is the worse choice. So feel free to update it.

> @comius Friendly ping. If you could clarify your preference regarding the fixedEnv/inheritedEnv precedence, I could finalize the PR ahead of your final review.

I agree current order gives you a bit more choice (i.e. having defaults possibly overridden from env) than the opposite (inherit but always override, where duplication doesn't make sense).